### DISCUSSION TITLE
Sh-esp32 Canbus Nema isolation

### DISCUSSION THREAD

**Discussion by g7wbd** [hatlabs/discussions#70] [General]
Having just ordered the SH-ESP32 Development Board, Downloaded the Kicad pcb files.
https://docs.hatlabs.fi/sh-esp32/

So Nema2000 is shrouded in mystery unless you pay to join. This Dev board has taken very special care of isolating the Canbus from the main board. is this a N2k thing (Backbone) ? or do we just compromise the isolation and then power board from same power source ?
How is the Backbone Isolated from the rest of the Boat. do we expect to have two or more independent power sources while interfacing to a Nema style boat ? 
Thanks
David

**Comment by mairas** [comment]
Hi David,

(First, it's NMEA, not Nema.)

I'm not sure I understand your questions. The N2K backbone is not isolated from the rest of the boat but the devices that connect to it need to have isolation at some point if they connect to other systems.

I've done a small writeup on NMEA 2000 isolation approaches: https://docs.hatlabs.fi/nmea2000/powering_devices/. Have a look at that page, maybe that will help? And possibly this one: https://docs.hatlabs.fi/nmea2000/

**Reply by g7wbd** [reply]
ah....... got name wrong .. LOL sorry that will happen dyslexia. but that I should have caught that one  :) 
not read the links yet  :) 
so is it ok to power ESP32 side from same power supply NMEA is offering, if I only require a small amount of current. ?


read links :) 
"Not permitted: SH-RPi powered via the NMEA 2000 and connected to external devices with a galvanic connection."
I had not considered galvanic connections....  hmmm need to read more 
Thankyou

**Reply by mairas** [reply]
Yes, absolutely, in most cases.

**Reply by psm12** [reply]
Hoping not to hijack this thread but I'm just looking at the same thing (powering sh-esp32 from the can/NMEA2000 bus). Looking at the schematic can this be achieved by linking j501 to j204. Schematic page 2 and 8 respectively, unpopulated pcb through holes labelled vlink on the silkscreen adjacent to the power and NMEA connectors? If so I'll populate them with header pins and add wire links. 
Thanks,
Paul

**Reply by psm12** [reply]
Matti,

I see the answer to my question above is actually documented in the sh-esp32 pages. I should have checked there first!

**Reply by kozy** [reply]
Hi everyone, its still not clear for me what is meant in the docs mentioned here. Explained is: 'Isolation meaning not sharing ground'. This is totally clear to me. This is for example galvanic isolation. 

Then there is:

_'OK: SH-RPi powered via the NMEA 2000 network and connected to external devices with an isolated connection such as Ethernet or isolated NMEA 0183.'_

(I assume isolated here is galvanic isolation)

And the other example: 

_'Not permitted: SH-RPi powered via the NMEA 2000 and connected to external devices with a galvanic connection. This setup will cause a lot of noise and may impact the reliability of the Raspberry Pi or other devices.'_

Galvanic connection meaning galvanic isolated connection there I believe. However to me this is then in contradiction. I dont think you mean `without` galvanic connection, no?

Also, since the SH-ESP-32 in my case will be powered from the same source, in the end they will share the same ground.

So what is best practice?

- Connect via a separate lead to the same source, so we get more of a star ground topology?
- Install my own galvanic isolation by getting a separate, isolated power supply. (outside of usb in my case)
- Supply power from N2K backbone is fine. (maybe/if there are low demands on current? I can't find where this is stated in the docs though)

Thanks for any help! :)

**Reply by mairas** [reply]
I'm not quite sure what you're planning to do and what devices you have, so I can't comment on the details, but the main point is, ground (and signals/voltages referencing ground such as a 3.3V line or resistance measurement!) should always have a star topology and no loops. Ground loops will both radiate noise and induce noise in the connected circuits.

The maximum current draw of any individual device on the NMEA 2000 network is 1 A. The SH-RPi limits its input current to ca. 0.8 A. It's still a lot of current, given the relatively thin AWG 22 wiring of a standard NMEA 2000 micro cabling, but it's well within the specs and works fine. I recommend having the Pi somewhat close to the power feed point on the network, though. If you place it at the end of a long backbone, there can be a substantial voltage drop.

**Reply by kozy** [reply]
Thanks for the explanation, I think I got it now :)

~~Regarding current, that should be okay. I was only confused since it was mentioned and might be related to ground loops, which it doesn't seem.~~ [It does matter and needs some thought.](https://en.wikipedia.org/wiki/Ground_loop_(electricity)#cite_ref-1)

The situation and my preliminary conclusion:

I want to read sensors on the engine with the SH-ESP32 (with engine top hat). I have two batteries, one lithium(house bank), one lead-acid(engine). 
They are not connected but as I read it, it might be good to connect the ground of both together in the center, then supply the SH-ESP32 with ground/power coming near this center point. Alternatively I could leave both floating and supply SH-ESP32 only from the lead acid but I don't like potential accidental draining.
Since the NMEA 2000 network on my (30ft) boat is rather short and can lay the backbone cable in a way that would be the optimal routing of a direct wire, I think that would be fine then. There is no special isolating safety circuitry bypassed if I connect the traces on the bottom of the SH-ESP32 pcb, I should just understand that ground is now shared with the NMEA 2000 network.

Details:

- 30ft polyester boat.
- Lithium battery with a non isolated step down converter supplying 12v to the NMEA 2000 network.
- Diesel engine with its own batteries for the dc starting engine. Not (purposely) sharing ground with the Lithium batteries.
- SH-ESP32 with an Engine Top Hat, connected to:
  - Diesel [alternator](https://www.dedieseldokter.nl/a-88694763/elektra-onderdelen/bukh-dv10-dv20-dv24-dv29-dv36-dv36-dv48-dynamo-008e8258-008e9525/#description) W terminal (I might charge the lithium in the future from this alternator as well)
  - [pressure](https://www.nauticgear.nl/bootaccessoires/bootonderdelen/motorinstrumenten/cn-boot-gevers/cn-druksensor-met-alarmpunt.html) and [temperature](https://www.nauticgear.nl/bootaccessoires/bootonderdelen/motorinstrumenten/cn-boot-gevers/cn-temperatuur-sensor.html) sensors for the diesel
  - i2c bilge [water level meter](https://www.kiwi-electronics.com/en/grove-water-level-sensor-10cm-for-arduino-9912)
  - [Davis vantage 6410](https://www.davis-europe.nl/product/davis-6410-anemometer-for-vantage-pro2-vantage-pro/)
- A [couple](https://www.raymarine.com/nl-nl/onze-producten/instrumenten/instrument-accessoires/itc-5-transducer-converter) of [other](https://openmarine.net/macarthur-hat) devices connected to the NMEA2000 network.
